metrics.go

```diff
@@ -15,55 +15,105 @@ type CertificateDesc struct {
 }
 
 type MetricsCollector struct {
-	certsMetric *prometheus.Desc
+	certsMetric  *prometheus.Desc
+	errorsMetric *prometheus.Desc
 
-	role           string
-	updates        chan VaultEvent
-	resourceEvents map[string]VaultEvent
-	resourcesMutex sync.RWMutex
+	role string
+
+	resourcesUpdates chan VaultEvent
+	resourceEvents   map[string]VaultEvent
+
+	errorUpdates  chan string
+	errorCounters map[string]int
+
+	metricsMutex sync.RWMutex
+}
+
+var errorUpdates chan string
+var errorUpdatesMutex sync.Mutex
+
+func setErrorUpdatesChan(ch chan string) {
+	errorUpdatesMutex.Lock()
+	defer errorUpdatesMutex.Unlock()
+
+	errorUpdates = ch
+}
+
+func getErrorUpdatesChan() chan string {
+	errorUpdatesMutex.Lock()
+	defer errorUpdatesMutex.Unlock()
+
+	return errorUpdates
 }
 
 func (m MetricsCollector) init() {
 	for {
 		select {
-		case event := <-m.updates:
+		case event := <-m.resourcesUpdates:
 			if event.Type == EventTypeFailure {
 				continue
 				// TODO: emit some nice metrics here
```

Comment:
We could log an error metric here now that we've got it implemented. What do you think? The label could be something like `update_failure`.

Reply:
Now that I think about it, we should not emit any metric here since if vault-sidekick failed to update a resource it means we have already emitted metrics for that in `vault.go`